pcp/Metric.c

```diff
@@ -165,14 +224,15 @@ bool Metric_fetch(struct timeval* timestamp) {
       pmFreeResult(pcp->result);
       pcp->result = NULL;
    }
-   int sts, count = 0;
+   int sts, count = 0, total;
    if (pcp->reconnect) {
       if (pmReconnectContext(pcp->context) < 0)
          return false;
       pcp->reconnect = false;
    }
+   total = (int) pcp->totalMetrics;
```

Comment:
```suggestion
   if (pcp->reconnect) {
      if (pmReconnectContext(pcp->context) < 0)
         return false;
      pcp->reconnect = false;
   }
   int sts, count = 0;
   int total = (int) pcp->totalMetrics;
```

Reduce scope of these variables